### PR TITLE
Miscellaneous patches

### PR DESCRIPTION
**Pull request by madebr**
This PR contains patches from all over the place. Changes are small, imho:
- Add `error_code` to `rwfs` namespace: some functions need it for the non-throwing version.
- Allow the configuration file to be passed by argument option
- `GameConfig` returns `rwfs::path` objects if paths are requested
- `GameData` and `LoaderIMG` accept `rwfs::path` objects
- Create strings and matrices in one go (less code)
- Remove some unused function definitions, function implementations and files
- Group the arguments thematically instead of one block

### CODE REVIEW

File: rwengine/src/engine/GameData.cpp [hunk 1]
```diff
@@ -695,12 +695,14 @@ float GameData::getWaveHeightAt(const glm::vec3& ws) const {
     return (1 + std::sin(engine->getGameTime() + (ws.x + ws.y) * WATER_SCALE)) *
            WATER_HEIGHT;
 }
-
-bool GameData::isValidGameDirectory(const std::string& path) {
-    if (path.empty()) {
+#include <boost/system/error_code.hpp>
```
Comment (danhedron) [inline rwengine/src/engine/GameData.cpp]: What does this include do here?

Comment (madebr) [reply rwengine/src/engine/GameData.cpp]: Rest over include. Nothing apparently.

File: rwgame/states/IngameState.cpp [hunk 2]
```diff
@@ -351,18 +351,7 @@ const ViewCamera& IngameState::getCamera(float alpha) {
...
-        m[0][2] = up.x;
-
-        m[1][0] = direction.y;
-        m[1][1] = right.y;
-        m[1][2] = up.y;
-
-        m[2][0] = direction.z;
-        m[2][1] = right.z;
-        m[2][2] = up.z;
+        glm::mat3 m(direction, right, up);
```
Comment (danhedron) [inline rwgame/states/IngameState.cpp]: This change totally jumbles the camera angles

File: rwlib/source/loaders/LoaderIMG.cpp [hunk 3]
```diff
@@ -5,9 +5,9 @@
 LoaderIMG::LoaderIMG() : m_version(GTAIIIVC), m_assetCount(0) {
 }
 
-bool LoaderIMG::load(const std::string& filename) {
-    auto baseName = filename;
-    auto extpos = filename.find(".img");
+bool LoaderIMG::load(const rwfs::path& filepath) {
+    auto baseName = filepath.string();
+    auto extpos = baseName.find(".img");
```
Comment (danhedron) [inline rwlib/source/loaders/LoaderIMG.cpp]: Could this be achieved with path::stem() instead?

File: rwgame/states/IngameState.cpp (hunk 2 above)
Comment (madebr) [reply rwgame/states/IngameState.cpp]: Oops, screw up on my part!

File: rwlib/source/loaders/LoaderIMG.cpp (hunk 3 above)
Comment (madebr) [reply rwlib/source/loaders/LoaderIMG.cpp]: I changed the code with `replace_extension`.
Too bad C++ does not have `with_extension` (like Python has with [with_suffix](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix))

File: rwgame/GameBase.cpp [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
 const std::string kBuildStr(kGitSHA1Hash, 8);
 const std::string kWindowTitle = "RWGame";
+const std::string kDefaultConfigFileName = "openrw.ini";
 constexpr int kWindowWidth = 800;
 constexpr int kWindowHeight = 600;
 
-GameBase::GameBase(Logger &inlog, int argc, char *argv[]) : log(inlog) {
+GameBase::GameBase(Logger &inlog, int argc, char *argv[]) :
+        log(inlog),
+        config(),
```
Comment (danhedron) [inline rwgame/GameBase.cpp]: Why put these empty constructors in the initaliser list?

Comment (madebr) [reply rwgame/GameBase.cpp]: I always try to be explicit about these things.

**Pull request merged**